### PR TITLE
Adds a unit test to prove that https://jira.duraspace.org/browse/DURA…

### PR DESCRIPTION
**Pull request by dbernstein**
…CLOUD-1131

was problem.  As it turns out,  the test confirmed that the retry logic works as expected.
However there was a corner case wherein a file that is re-added to the ChangedList while
the previously added file could possibly not receive the expected number of retries in the
event that the reserved file is unreserved.

Addresses: https://jira.duraspace.org/browse/DURACLOUD-1131

### CODE REVIEW

File: synctool/src/main/java/org/duracloud/sync/mgmt/ChangedFile.java [hunk 1]
```diff
@@ -35,10 +35,17 @@ public void incrementSyncAttempts() {
         syncAttempts++;
     }
     
+    /**
+     * Removes a previously reserved from the ChangedList.
```
Comment (bbranan) [inline synctool/src/main/java/org/duracloud/sync/mgmt/ChangedFile.java]: Should this be:  "Removes a previously reserved **file** from the ChangedList."
